src/backend/distributed/utils/maintenanced.c

```diff
@@ -644,8 +644,18 @@ CitusMaintenanceDaemonMain(Datum main_arg)
 				 */
 				lastShardCleanTime = GetCurrentTimestamp();
 
-				bool waitForCleanupLock = false;
-				numberOfDroppedShards = TryDropMarkedShards(waitForCleanupLock);
+				Oid distPlacementId = DistPlacementRelationId();
```

Comment:
I have manually tested this but it might be good to add a test for this if it is not super hard